File: java/infeccion/main/java/com/public/decompiler.java
package com.ejemplo;

import java.io.IOException;

public class Decompiler {
    public static void decompileApk(String apkFile, String outputDir) {
        try {
            Process process = new ProcessBuilder("apktool", "d", apkFile, "-o", outputDir)
                .inheritIO()
                .start();
            process.waitFor();
            System.out.println("APK decompilado correctamente en " + outputDir);
        } catch (IOException | InterruptedException e) {
            System.out.println("Error al decompilar el APK: " + e.getMessage());
        }
    }
}

File: python/infeccion/decompiler.py
import subprocess

def decompile_apk(apk_file, output_dir):
    result = subprocess.run(['apktool', 'd', apk_file, '-o', output_dir], capture_output=True, text=True)
    if result.returncode == 0:
        print(f"APK decompilado correctamente en {output_dir}")
    else:
        print(f"Error al decompilar el APK: {result.stderr}")
